### PR TITLE
`Start-Transcript` Add option to minimise transcript header

### PR DESCRIPTION
**Pull request by ghost**
## PR Summary

Add a switch parameter `UseMinimalHeader` to `Start-Transcript`. When used, a transcript will contain a short header like the following, instead of the usual header containing version information.

```
**********************
PowerShell transcript start
Start time: 20181201203347
**********************
```

This partly addresses #8359.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by iSazonov** [comment]
@lukexjeremy Thanks for your contribution!
You could look `Start-Transcript.Tests.ps1` for tests.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs [hunk 1]
```diff
@@ -137,6 +137,15 @@ public SwitchParameter IncludeInvocationHeader
             get; set;
         }
 
+        /// <summary>
+        /// Gets or sets whether to exclude transcript header.
+        /// </summary>
+        [Parameter]
+        public SwitchParameter MinimalTranscriptHeader
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs]: I suggest to use `-UseMinimalHeader`

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs [hunk 2]
```diff
@@ -221,7 +230,7 @@ protected override void BeginProcessing()
 
                 System.Management.Automation.Remoting.PSSenderInfo psSenderInfo =
                     this.SessionState.PSVariable.GetValue("PSSenderInfo") as System.Management.Automation.Remoting.PSSenderInfo;
-                Host.UI.StartTranscribing(effectiveFilePath, psSenderInfo, IncludeInvocationHeader.ToBool());
+                Host.UI.StartTranscribing(effectiveFilePath, psSenderInfo, IncludeInvocationHeader.ToBool(), MinimalTranscriptHeader.ToBool());
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs]: You can pass in the value by `UseMinimalHeader.IsPresent`

File: src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs [hunk 3]
```diff
@@ -461,45 +461,58 @@ private void LogTranscriptHeader(System.Management.Automation.Remoting.PSSenderI
...
-                    psVersionInfo.ToString().TrimEnd()
-                    );
+            string line;
+            if (minimalTranscriptHeader)
+            {
+                line =
+                    string.Format(
+                        Globalization.CultureInfo.InvariantCulture,
+                        InternalHostUserInterfaceStrings.MinimalTranscriptPrologue,
+                        DateTime.Now);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs]: When `minimalTranscriptHeader` is true, there is no need to calculate the unused information like `versioninfo` and `username`. So it's better to organize the code this way:
```
string line;
if (useMinimalHeader)
{
    line = ...
}
else
{
    string username  ...
    string runAsUser ...
    ...
    line = ...
}
```

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

**Review by PaulHigin — APPROVED** [review]

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

**Comment by daxian-dbw** [comment]
@lukexjeremy On another thought, `-UseShortHeader` might be a better choice as you only have two options: the regular header and the short header, while `minimal` seems suggesting you have at least 3 options for the header. What do you think?
At the meantime, let me consult with the team members and reply back what they think.

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs]: What is best practice? Why do we use a verb on first place? Can we use more short name `MinimalHeader`, `ShortHeader`, or `BriefHeader`?

Comment (daxian-dbw) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs]: Readability (make the command use more descriptive) and preceding examples? There are many cmdlets with `-Use*` parameters.

**Review by daxian-dbw — APPROVED** [review]

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs]: I'd give priority to IntelliSense (unique first char).

Comment (daxian-dbw) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs]: It's not like you are adding the same prefix to multiple parameters of the same command. You will still get the unique first char when completing parameter for a specific command.

**Comment by daxian-dbw** [comment]
@lukexjeremy Thank you for opening the doc issue. I forgot that myself :)

**Pull request merged**